json-rpc/src/tests.rs

```diff
@@ -69,44 +72,53 @@ fn test_json_rpc_protocol() {
     assert_eq!(fetch_error(resp), -32000);
 }
 
+fn setup_db(
+    blocks: Vec<(Vec<TransactionToCommit>, LedgerInfoWithSignatures)>,
+) -> (Arc<LibraDB>, BTreeMap<AccountAddress, AccountResource>) {
+    // set up db
+    let tmp_dir = TempPath::new();
+    let db = Arc::new(LibraDB::new(&tmp_dir));
+    let mut version = 0;
+    let mut all_accounts = BTreeMap::new();
+    let mut all_txns = vec![];
+
+    for (txns_to_commit, ledger_info_with_sigs) in &blocks {
+        db.save_transactions(
+            &txns_to_commit.clone(),
+            version,
+            Some(&ledger_info_with_sigs.clone()),
+        )
+        .unwrap();
+        version += txns_to_commit.len() as u64;
+        let mut account_states = HashMap::new();
+        // Get the ground truth of account states.
+        txns_to_commit.iter().for_each(|txn_to_commit| {
+            account_states.extend(txn_to_commit.account_states().clone())
+        });
+
+        // Record all account states.
+        for (address, blob) in account_states.iter() {
+            all_accounts.insert(address.clone(), AccountResource::try_from(blob).unwrap());
+        }
+
+        // Record all transactions.
+        all_txns.extend(
+            txns_to_commit
+                .iter()
+                .map(|txn_to_commit| txn_to_commit.transaction().clone()),
+        );
+    }
+
+    (db, all_accounts)
+}
+
```

Comment:
Do we just want to an mock db?  This will require write to disk.  which may be too heavy for an unittest.

Reply:
yeah - a MockDB doesn't exist right now, so I can make a separate diff for this